xdsl/dialects/memref.py

```diff
@@ -484,20 +485,33 @@ def verify_(self) -> None:
                 "Global initial value is expected to be a "
                 "dense type or an unit attribute"
             )
+        if self.alignment is not None:
+            assert isa(self.alignment, IntegerAttr)
+            test = self.alignment.value.data
+            # test if is power of 2 with binary magic
+            if not ((test & (test - 1) == 0) and test != 0):
+                raise VerifyException(
+                    f"Alignment attribute {self.alignment.value.data} is not a power of 2"
+                )
 
     @staticmethod
     def get(
         sym_name: StringAttr,
         sym_type: Attribute,
         initial_value: Attribute,
         sym_visibility: StringAttr = StringAttr("private"),
+        alignment: int | AnyIntegerAttr | None = None,
```

Comment:
If it has to be i64 then we should probably use `IntegerAttr[IntegerType]`

Reply:
That makes more sense! thanks!